opensrp-fp/src/main/java/org/smartregister/fp/common/util/Utils.java

```diff
@@ -1120,4 +1118,61 @@ public static void openMECWheelApp(Context context) {
         }
         context.startActivity(intent);
     }
+
+    public static LocalDate getNextContactVisitDate(JSONObject jsonObject) {
+        String methodExit = getJsonFieldValue(jsonObject, "step7", "method_exit");
+
+        DateTimeFormatter pattern = DateTimeFormat.forPattern(("dd-MM-yyyy"));
+        LocalDate nextContactDate = null;
+        switch (methodExit) {
+            case "cu_iud":
+            case "lng_iud":
+                nextContactDate = LocalDate.parse(getJsonFieldValue(jsonObject,"step7", "iud_insertion_date"), pattern);
+                break;
+
+            case "dmpa_im":
+            case "dmpa_sc":
+            case "net_en":
+                nextContactDate = LocalDate.parse(getJsonFieldValue(jsonObject,"step7", "last_injection_date"), pattern);
+                break;
+
+            case "pop":
+            case "coc":
+            case "patch":
+            case "cvr":
+            case "pvr":
+                nextContactDate = LocalDate.parse(getJsonFieldValue(jsonObject,"step1", "visit_date"), pattern);
+                break;
+
+            case "male_sterilization":
+            case "female_sterilization":
+                nextContactDate = LocalDate.parse(getJsonFieldValue(jsonObject,"step7", "sterilization_date"), pattern);
+                break;
+
+            default:
+                break;
+        }
+
+        return nextContactDate;
+    }
+
+    private static String getJsonFieldValue(JSONObject jsonObject, String stepKey, String fieldKey) {
+        String value = "";
+
+        try {
+            JSONArray jsonArray = jsonObject.getJSONObject(stepKey).getJSONArray("fields");
```

Comment:
Please move the hard coded string to a constants file

Reply:
Done